index.html

```diff
@@ -0,0 +1,300 @@
+<input type="text" id="heading_text" value="-1" maxlength="4" size="4" /></br>
+Costing: <select id="costing">
+  <option>none</option>
+  <option selected>auto</option>
+  <option>pedestrian</option>
+  <option>bicycle</option>
+</select></br>
+
+    </div>
+    <div id="sidebar" style="float:right; width:25%">
+      <pre id="properties"></pre>
+    </div>
+  </div>
+
+
+  <script>
+    //update the properties
+    var pretty = {
+      replacer: function(match, pIndent, pKey, pVal, pEnd) {
+        var key = '<span class=json-key>';
+        var num = '<span class=json-number>';
+        var str = '<span class=json-string>';
+        var con = '<span class=json-constant>';
+        var r = pIndent || '';
+        if (pKey)
+          r = r + key + pKey.replace(/: +$/g, '') + '</span>: ';
+        if (pVal) {
+          var which = pVal[0] == '"' ? str : (isNaN(pVal) ? num : con);
+          r = r + which + pVal + '</span>';
+        }
+        return r + (pEnd || '');
+      },
+      print: function(obj) {
+        var line = /^( *)("[\w]+": )?("[^"]*"|[\w.+-]*)?([,[{]|\[\],?|{},?)?$/mg;
+        var html = JSON.stringify(obj, null, 2)
+          .replace(/&/g, '&amp;').replace(/\\"/g, '&quot;')
+          .replace(/</g, '&lt;').replace(/>/g, '&gt;');
+        var y = html.replace(line, pretty.replacer);
+        return y;
+      }
+    };
+
+    //keep an object around to parse protos with
+    var SpeedTile;
+    protobuf.load('https://s3.amazonaws.com/speed-extracts/speedtile.proto', function(err, root) {
+      //couldnt parse proto definition
+      if(err) throw err;
+      //get the message type
+      SpeedTile = root.lookupType('SpeedTile');
+    });
+
+    //download a file as a byte buffer
+    var getByteBufferUrl = function(url, callback) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', url, true);
+      xhr.responseType = 'arraybuffer';
+      xhr.onload = function(e) {
+        if(this.status == 200) {
+          //let the callback do something with the blob
+          callback(new Uint8Array(this.response));
+        }
+      };
+      xhr.send();
+    }
+
+    var LEVEL_BITS = 3;
+    var TILE_INDEX_BITS = 22;
+    var SEGMENT_INDEX_BITS = 21;
+    var LEVEL_MASK = Math.pow(2, LEVEL_BITS) - 1;
+    var TILE_INDEX_MASK = Math.pow(2, TILE_INDEX_BITS) - 1;
+    var SEGMENT_INDEX_MASK = Math.pow(2, SEGMENT_INDEX_BITS) - 1;
+    var VALHALLA_TILES = [
+      { level: 2, size: 0.25 },
+      { level: 1, size: 1.0 },
+      { level: 0, size: 4.0 }
+    ];
+
+    //js has 53bits of precision but if you need to get values out
+    //that are passed the 32nd bit, you cant use bitwise operators
+    function get_level(segment_id){
+      return segment_id & LEVEL_MASK;
+    }
+    function get_tile_index(segment_id){
+      return (segment_id >> LEVEL_BITS) & TILE_INDEX_MASK;
+    }
+    function get_segment_index(segment_id){
+      return (segment_id / Math.pow(2, LEVEL_BITS + TILE_INDEX_BITS)) & SEGMENT_INDEX_MASK;
```

Comment:
thanks - yes I ran into this very early on and ended up asking a library to do it for me. I'll use this and replace the lib